### ISSUE TITLE
Engine Updates – missing updaters

### ISSUE THREAD

**Issue by theflow** [depfu/depfu#43]
Updaters are what we call the code handling a single file, like a `.ruby-version` or `.circle/config.yml`.  They know where to find the version and how to update it.

As you can imagine this can be a bit brittle, but in most cases we're quite confident we can make it work well.

We started with a small list of files, so if you're specifying the version of your Ruby/Node.js/Elixir in a file we don't support, we want to know and quickly add support to it.

So far we support

- `.ruby-version`
- `.node-version`
- `.nvmrc`
- `.exenv-version`
- `.tool-versions`
- `Gemfile` and `Gemfile.lock`
- `mix.exs`
- `package.json`
- `.circleci/config.yml`
- `.travis.yml`
- `Dockerfile`
- `docker-compose.yml`

👉 Please just add a comment for any file that is missing and we'll take a look right away.

**Comment by benedikt** [comment]
Fantastic new feature! 🔥 I'm really excited about this 👍 

We're using both `.ruby-version` and the `Gemfile` to specify the Ruby version, but we're using a relaxed definition in the `Gemfile` like this:

```ruby
ruby '~> 2.6.0'
```

The pull request to upgrade to Ruby 2.7.1 updated both the `.ruby-version` and the `Gemfile.lock` but didn't update the version specification in `Gemfile`.

**Comment by halfbyte** [comment]
@benedikt Good catch, we only update the Gemfile if it's using a simple version requirement. I'm working on a fix. Thanks for the early testing!

**Comment by Siilwyn** [comment]
Do you support the `package.json` [Volta pinned fields](https://docs.volta.sh/guide/understanding#pinning-node-engines)?

**Comment by rmehner** [comment]
Changed my `.node-version` by hand to the version depfu wanted to upgrade to (since I also had to adjust versions in the Dockerfiles) and depfu didn't autoclose its own PR because the version was already applied.

No biggie, just noticed. Nice feature all around! 👏

**Comment by matthewlehner** [comment]
This is an incredible feature! Just tried it for the first time and was very surprised that it correctly updated our Dockerfile!

The one thing that this missed was a workflow for Github actions.

We're using Elixir with GitHub actions – it missed `.github/workflows/*.yml` with the https://github.com/erlef/setup-elixir action, like this:

```yaml
      - name: Setup elixir
        uses: erlef/setup-elixir@v1
        with:
          elixir-version: 1.12.1 # Define the elixir version [required]
          otp-version: 24.0.2 # Define the OTP version [required]
```

I recognize that this is probably a tough one to support, but since you're supporting CircleCI, I thought I'd ask. If you've got the ability to add this, it'd be incredible.

**Comment by bmulholland** [comment]
Rubocop has an option for `TargetRubyVersion` in its `.rubocop.yml` that needs to be updated too.

**Comment by CorySanin** [comment]
this pull request https://github.com/CorySanin/ffa-tycoon/pull/13 is proposing I downgrade from the latest LTS version of node. Neat feature though.

edit: omg sorry for the thread bump 😬

**Comment by viktorianer** [comment]
Support for `Dockerfile` works, bot how abut related files like `Dockerfile.dev` and so on?

Also, if in `package.json` I have something like `"node": "~18.x",` I would like to have upgrade to `"node": "~20.x"` and not `"node": "^20.6.1"`. The same for all other files and engines.

What about an option to upgrade only to the latest LTS version (Node.js 18.18.0 LTS), rather to the latest available version (Node.js 20.7.0 Current)?

**Comment by bvogel** [comment]
Our dockerfiles aren't touched bc their naming does not fit the general naming schema. Ours are
- base.Dockerfile
- test.Dockerfile
- Dockerfile.local

would it be feasible to add some kind of `.depfurc`-file to add these files to or should the glob be extended to `*Dockerfile*`?

**Comment by akostadinov** [comment]
[Open Container Initiative](https://opencontainers.org/) file name is `Containerfile`. Would be useful to support that as well.

**Comment by klausmeyer** [comment]
@theflow Could you also add `.devcontainer/Dockerfile` please?

Example:

```
FROM ghcr.io/rails/devcontainer/images/ruby:3.4.4
```

**Comment by rnestler** [comment]
In the `rails new` generated `Dockerfile` the version is generated as follows:
```Dockerfile
# Make sure RUBY_VERSION matches the Ruby version in .ruby-version
ARG RUBY_VERSION=3.3.8
FROM docker.io/library/ruby:$RUBY_VERSION-slim AS base
``` 

Is it possible to add support for changing the `RUBY_VERSION` arg default value in the engine updates?

**Comment by rnestler** [comment]
Inside a `.gitlab-ci.yml` the docker image tag:

<img width="751" height="148" alt="Image" src="https://github.com/user-attachments/assets/a113c503-0195-42e7-9ee9-323239791a49" />

**Comment by rnestler** [comment]
Inside a `Dockerfile` a `NODE_VERSION` assignment:

<img width="1032" height="443" alt="Image" src="https://github.com/user-attachments/assets/6ca9781d-3aa3-46d5-a3e9-f0f1dd25d16b" />

**Comment by 12joan** [comment]
> Support for `Dockerfile` works, bot how abut related files like `Dockerfile.dev` and so on?

I'm also experiencing this. Perhaps a good heuristic would be any file with `Dockerfile` in the name (case-insensitive) that contains a line starting with `FROM`?
